caps/capabilities_test.go

```diff
@@ -0,0 +1,80 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package caps
+
+import (
+	"testing"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/ubuntu-core/snappy/testutil"
+)
+
+func Test(t *testing.T) {
+	TestingT(t)
+}
+
+type CapabilitySuite struct{}
+
+var _ = Suite(&CapabilitySuite{})
+
+func (s *CapabilitySuite) TestValidateName(c *C) {
+	c.Assert(ValidateName("name with space"), Equals, false)
+	c.Assert(ValidateName("name-with-trailing-dash-"), Equals, false)
+	c.Assert(ValidateName("name-with-3-dashes"), Equals, true)
+	c.Assert(ValidateName("name"), Equals, true)
+}
+
+func (s *CapabilitySuite) TestAdd(c *C) {
+	repo := NewRepository()
+	cap := &Capability{"name", "label", FileType}
+	c.Assert(repo.Names(), Not(testutil.Contains), cap.Name)
+	err := repo.Add(cap)
+	c.Assert(err, IsNil)
+	c.Assert(repo.Names(), testutil.Contains, cap.Name)
```

Comment:
You could do a "c.Assert(repo.names(), DeepEquals, []string{cap.Name})" test here. This woudl be slightly better I think because it woudl test that the name is not added twice (well, unlikely with the current data structures but :)